reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/Buffer.kt

```diff
@@ -57,8 +57,8 @@ fun <T> Observable<T>.buffer(
 fun <T, S> Observable<T>.buffer(
     opening: Observable<S>,
     closing: (S) -> Completable,
-    limit: Long = Long.MAX_VALUE,
+    limit: Int = Int.MAX_VALUE,
     restartOnLimit: Boolean = false
 ): Observable<List<T>> =
-    window(opening = opening, closing = closing, limit = limit, restartOnLimit = restartOnLimit)
+    window(opening = opening, closing = closing, limit = limit.toLong(), restartOnLimit = restartOnLimit)
```

Comment:
Are you going to update `window` too?

Reply:
Window looks fine with long, it doesn't have any buffers, just `Long` counters. RxJava also uses `Long` for windows - http://reactivex.io/RxJava/javadoc/io/reactivex/Observable.html#window-long-java.util.concurrent.TimeUnit-io.reactivex.Scheduler-long-boolean-